clients/src/main/java/org/apache/kafka/clients/consumer/internals/PartitionAssignor.java

```diff
@@ -56,7 +57,7 @@
      * @return A map from the members to their respective assignment. This should have one entry
      *         for all members who in the input subscription map.
      */
-    Map<String, Assignment> assign(Cluster metadata, Map<String, Subscription> subscriptions);
+    Map<String, Assignment> assign(Cluster metadata, Map<MemberInfo, Subscription> subscriptions);
```

Comment:
Another meta comment: this is a public change that would enforce all implementers to make code changes. I'm wondering if we could instead piggy-back the optional instance.id into subscription, so that only those who are indeed aware of the KIP-345 and want to leverage it in their own assignor implementation then need to make code changes, but others stay the same.